appserver/payara-appserver-modules/microprofile/opentracing/src/main/java/fish/payara/microprofile/opentracing/jaxrs/JaxrsContainerRequestTracingExceptionMapper.java

```diff
@@ -58,24 +57,18 @@
     @Override
     public Response toResponse(Throwable exception) {
         StatusType status = Response.Status.INTERNAL_SERVER_ERROR;
-        URI location = null;
 
         // Get the status and location if available
         if (exception instanceof WebApplicationException) {
-            Response response = ((WebApplicationException) exception).getResponse();
-            location = response.getLocation();
-            status = response.getStatusInfo();
+            return ((WebApplicationException) exception).getResponse();
         }
 
         Response.ResponseBuilder responseBuilder = Response.status(status);
-        if (location != null) {
-            responseBuilder.location(location);
-        }
+
         // If the status is a server error, attach it as an entity, otherwise just return a response
         if (status.getFamily() == Response.Status.Family.SERVER_ERROR) {
```

Comment:
I'm not sure this fix works. Does it pass the OpenTracing TCK? If not, I believe that the response entity and status still need changing. Everything else should be carried across.